### DISCUSSION TITLE
Data Object Issue In Perspective

### DISCUSSION THREAD

**Discussion by KasthuriAnandhan** [pimcore/pimcore#17606] [Q&A / Support]
Hi @kingjia90 ,

I created perspectives and custom views in my Pimcore environment. I assigned a role to a user and logged in using those credentials. The perspective contains data objects from four classes. When I open the 'AllProducts' class data object, I encounter the following error. However, when I open data objects from the other three classes, I don't face any issues. I only need 'List' and 'View' permissions for the 'AllProducts' class data objects, so I enabled only those permissions. The 'AllProducts' class includes object bricks and some one-to-one relation attributes. Could this be causing the issue I’m facing?

![Screenshot 2024-09-12 182244](https://github.com/user-attachments/assets/2ee18541-219d-4226-b903-c919470654ae)

**Comment by ravi-vegda** [comment]
you can check class defined in the perspective for that object, might be that cause issue

**Comment by fszenborn** [comment]
check auto save function maybe thats an issue
